chill-protobuf/src/main/java/com/twitter/chill/protobuf/ProtobufSerializer.java

```diff
@@ -69,6 +69,9 @@ public void write(Kryo kryo, Output output, Message mes) {
   public Message read(Kryo kryo, Input input, Class<Message> pbClass) {
```

Comment:
@regadas Wanted to get your thoughts on handling `null` in the serializer.

I was thinking maybe we could serialize nulls to a magic number (`-1`) which would look like:
```java
@Override
public void write(Kryo kryo, Output output, Message mes) {
    if (mes == null) {
        output.writeInt(-1, true);
    }
    ...
}
```
and 

```java
@Override
public Message read(Kryo kryo, Input input, Class<Message> pbClass) {
    try {      
        int size = input.readInt(true);
        if (size == -1) {
            return null;
        }
        ...
    } catch (Exception e) {
        throw new RuntimeException("Could not create " + pbClass, e); 
    }
}
```

Reply:
since kryo dispatches serialization based on the `.getClass` I think this code will never really be called on `null` in practice.

Can you show a test that fails now just using `Kyro` not the serializer directly?

Reply:
@johnynek you're right, I cannot get the test to fail inside the serializer with `null`. However, the `com.esotericsoftware.kryo.Serializer` base class has a `acceptsNull` field that's injected through the constructor. Hence, depending on how it's initialized, the serializer might get a `null` object?